src/main/java/core/basesyntax/SalaryInfo.java

```
@@ -1,7 +1,65 @@
+            resultInfo.append(dateFrom).append(" - ").append(dateTo)
+                    .append(System.lineSeparator());
+        } catch (NullPointerException | DateTimeException e) {
+            System.out.println("Date should be not NULL! and be in \"dd.MM.yyyy\" format");
+            System.out.println(e.getMessage());
+        }
+
+        // data and names not Null
+        if (data == null || names == null) {
+            throw new NullPointerException("Names and data arrays on input must be not null!");
+        }
+
+        // look for each name in names
+        for (String name : names) {
+            int salary = 0;
+            for (String employee : data) {
+                // creating array of Strings by each record
+                String [] employeeInfo = employee.split(" ");
+                if (employeeInfo.length != 4) {
+                    throw new IndexOutOfBoundsException("Wrong input data! "
+                            + "data must be the array of Strings such format :"
+                            + " 25.04.2019 John 60 50");
+                }
+
+                // if our date between input dates then take calculate salary
+                try {
+                    LocalDate parsedEmployeeDate = LocalDate.parse(employeeInfo[0], dateFormatter);
+                    if (parsedEmployeeDate.isBefore(parsedDateTo.plusDays(1))
+                            && parsedEmployeeDate.isAfter(parsedDateFrom.minusDays(1))) {
+                        if (name.equals(employeeInfo[1])) {
+                            salary += Integer.parseInt(employeeInfo[2])
+                                    * Integer.parseInt(employeeInfo[3]);
+                        }
+                    }
+                } catch (NullPointerException | DateTimeException e) {
+                    System.out.println("Date should be not NULL! and be in \"dd.MM.yyyy\" format");
+                    System.out.println(e.getMessage());
```

This `try-catch` block catches `NullPointerException` and `DateTimeException`, but it only prints the error message to the console. This doesn't help to resolve the problem and may hide potential issues. It's better to remove this `try-catch` block and let the exception propagate to the caller method, or handle it in a more meaningful way.